create-k8s-chained-sessions.sh

```diff
@@ -6,63 +6,92 @@
 # Do not run this script more than once without resetting your Leapp
 # instance beforehand.
 
+. ./utils.sh
+
 # global variables
-declare PROFILE_ID
-declare CHAINED_SESSION_IDS="name,id\n"
 declare REGION='us-east-1'
 
 ###### FUNCTIONS ######
 #
 # function to create a chained leapp session given a parent session id
 # Args:
 # 1: parent session id name from leapp
-# appends new session id from the new chained session to CHAINED_SESSION_IDS
+# 2: role name to use for the chained session
 function createLeappSession {
+    green_echo "creating chained session for $1 with role $2"
     parent_session_name=$1
-    chained_session_name="chained-from-${parent_session_name}"
-    echo "starting session for ${parent_session_name} to get role arn"
-    # this has funky piping because `--filter` is a fuzzy lookup and `panorama-k8s-playground` fuzzy matches `panorama-k8s-playground-2` as the first result
-    parent_session_id=$(leapp session list -x --filter="Session Name=${parent_session_name}" --no-header | sort -k2 | sed -n 1p | awk '{print $1}')
-    # start leapp session
-    leapp session start --sessionId $parent_session_id
-    # call to aws to get the role arn for `TerraformRole`
-    role_arn=$(aws iam get-role --role-name TerraformRole --query Role.Arn | tr -d '"')
-    # stop the leapp session
-    leapp session stop --sessionId $parent_session_id
+    parent_role_name=$2
+    chained_role_name=$3
+    # check if the parent session exists for the role. We do this because
+    # regular developers won't have the AWSAdministratorAccess role, so we
+    # don't want to create a chained session for them.
+    parent_session_id=$(leappSessionId "$parent_session_name" "$parent_role_name")
+    if [[ -z "${parent_session_id}" ]]; then
```

Comment:
What's the relationship between the parent session and the AWSAdministratrAccess role?

Are the non-admin sessions not chained, meaning only admins would ever need to run this?

Reply:
Regular developers will have the parent session for `PanoramaK8sDefaultEngineering` role, `AWSAdministratorAccess` is an example of a case where the parent doesn't exist, so we just skip it.

Reply:
Ah is it just to avoid erroring out when folks without access to that role run the script?